src/Chainweb/RestAPI.hs

```diff
@@ -224,15 +228,27 @@ chainwebApplication
     -> Maybe (MiningCoordination logger cas)
     -> HeaderStream
     -> Application
-chainwebApplication v dbs mr hs =
-    chainwebCors . someServerApplication $ someChainwebServer v dbs mr hs
+chainwebApplication v dbs mr hs
+    = chainwebTime
+    . chainwebCors
+    . someServerApplication
+    $ someChainwebServer v dbs mr hs
 
 -- Simple cors with actualy simpleHeaders which includes content-type.
 chainwebCors :: Middleware
 chainwebCors = cors . const . Just $ simpleCorsResourcePolicy
     { corsRequestHeaders = simpleHeaders
     }
 
+chainwebTime :: Middleware
+chainwebTime app req resp = app req resp'
+  where
+    resp' res = do
+        timestamp <- realToFrac <$> getPOSIXTime :: IO Double
+        resp $ mapResponseHeaders
+            ((:) ("X-Chainweb-Time", B8.pack (printf "%.0f" timestamp)))
```

Comment:
a) avoid `Double` for timestamps -- we already have a dep on the clock package so here I would prefer http://hackage.haskell.org/package/clock-0.8/docs/src/System.Clock.html#getTime with CLOCK_REALTIME_COARSE -- easy to convert the resulting `TimeSpec` to `Ratio Int64` for `printf`, or do the job by hand with `div` on the `nsec` field of the timespec

b) the header should be called "X-Server-Timestamp" (and the bike shed should be blue)

Reply:
It seems that `CLOCK_REALTIME_COARSE` is available only on linux (or at least it isn't obvious from the package code if it is also available on other platforms).

There's also no code for easy detection or graceful fallback available. So, I'll stick with `Realtime`.

I don't think that the `RealTime` clock is (much) faster than `getPOSIXTime` from `time` and we already use `time` in the same code path. But happy to use clock, instead.